src/RequestHeaders.cs

```diff
@@ -35,7 +35,9 @@ public class RequestHeaders : IDictionary<string,IEnumerable<string>> {
     /// <inheritdoc/>
     public bool IsReadOnly => false;
     /// <inheritdoc/>
+#pragma warning disable CS8603 // Possible null reference return. //Can't change signature of overriden method implementation
```

Comment:
Maybe return Emurable.Empty<string>() instead?

Reply:
As this is an indexer, I worth itould maybe prefer to maybe throw an `KeyNotFoundException` as returning an empty array would imply that the header exists but just without values. Either way we go, I think its important to highlight that this would however break the current expectations we set out earlier [here](https://github.com/microsoft/kiota-abstractions-dotnet/blob/b0cd59539aca7df88d203fd5cae6ca1b4323db66/Microsoft.Kiota.Abstractions.Tests/RequestHeadersTests.cs#L53).

Reply:
I believe that the only user of the indexer will be a client app developer. As far as I can remember the generated code only leverages the Add method (first one above), and the request adapter lists all the entries.
Dictionary throwing a key not found exception in that case, let's get closer to that implementation for consistency.

Reply:
Resolved in latest updates